File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;               // no -1
        for (int i = 0; i < (int)nums.size(); ++i) {
            int need = target - nums[i];
            auto it = hash.find(need);             // no insertion
            if (it != hash.end()) {
                return {it->second, i};            // return the pair
            }
            hash[nums[i]] = i;                     // insert after check
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start=0
        end=len(numbers)
        indices=[]

        while start<end:
            if(target==numbers[start]+numbers[end-1]):
                return [start+1,end]
            elif(target<numbers[start]+numbers[end-1]):
                end-=1
            else:
                start+=1
        
        return []
